src/main/java/ch/saunah/saunahbackend/controller/BookingController.java

```diff
@@ -39,15 +35,49 @@ public class BookingController {
     @Autowired
     private UserService userService;
 
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private MailService mailService;
+
     @Operation(description = "Creates a new booking.")
     @PostMapping(path = "bookings")
     public @ResponseBody
     ResponseEntity<BookingResponse> createBooking(@RequestBody BookingBody bookingBody, Principal principal) throws Exception {
         User currentUser = userService.getUserByMail(principal.getName());
         Booking booking = bookingService.addBooking(bookingBody, currentUser.getId());
+        mailService.sendAdminOpenedBookingMail(userRepository.findByRole(UserRole.ADMIN), booking);
+        mailService.sendUserOpenedBookingMail(userService.getUser(booking.getUserId()).getEmail(), booking, booking.getBookingPrice(), booking.getBookingSauna());
         return ResponseEntity.ok(new BookingResponse(booking));
     }
 
+    @Operation(description = "Returns the bookingPrice with the ID specified.")
+    @GetMapping(path = "bookings/{id}/price")
+    public @ResponseBody
+    ResponseEntity<BookingPriceResponse> getBookingPrice(@PathVariable(value = "id", required = true) Integer id, Principal principal) throws AuthenticationException {
+        Booking booking = bookingService.getBooking(id);
+        BookingPrice bookingPrice = bookingService.getBookingPrice(id);
+        User user = userService.getUserByMail(principal.getName());
+        if (booking.getUserId() == user.getId() || UserRole.ADMIN.equals(user.getRole())) {
+            return ResponseEntity.ok(new BookingPriceResponse(bookingPrice));
+        }
+        throw new AuthenticationException("user is not authenticated to view this booking");
+    }
+
+    @Operation(description = "Returns the bookingPrice with the ID specified.")
+    @GetMapping(path = "bookings/{id}/sauna")
+    public @ResponseBody
+    ResponseEntity<BookingSaunaResponse> getBookingSauna(@PathVariable(value = "id", required = true) Integer id, Principal principal) throws AuthenticationException {
+        Booking booking = bookingService.getBooking(id);
+        BookingSauna bookingSauna = bookingService.getBookingSauna(id);
```

Comment:
pass bookingPriceId from booking, id is not right when calling booking price because it can differ from id